eth/tracers/native/call.go

```diff
@@ -137,10 +150,42 @@ func (t *callTracer) CaptureEnd(output []byte, gasUsed uint64, _ time.Duration,
 
 // CaptureState implements the EVMLogger interface to trace a single step of VM execution.
 func (t *callTracer) CaptureState(pc uint64, op vm.OpCode, gas, cost uint64, scope *vm.ScopeContext, rData []byte, depth int, err error) {
+	// Only logs need to be captured via opcode processing
+	if !t.config.WithLog {
+		return
+	}
+	// Avoid processing nested calls when only caring about top call
+	if t.config.OnlyTopCall && depth > 0 {
+		return
+	}
+	switch op {
+	case vm.LOG0, vm.LOG1, vm.LOG2, vm.LOG3, vm.LOG4:
+		size := int(op - vm.LOG0)
+
+		stack := scope.Stack
+		stackData := stack.Data()
+		stackPop := func() uint256.Int {
+			ret := stackData[len(stackData)-1]
+			stackData = stackData[:len(stackData)-1]
+			return ret
+		}
+
+		mStart := stackPop()
+		mSize := stackPop()
```

Comment:
All in all, I think this is fine. However -- a tracer should not modify data-structures, such as the `stack` inside the `scope`. This code doesn't actually modify it, I think, but that is not obvious. 
Why not just do
```
		mStart := stackData[len(stackData)-1]
		mStart := stackData[len(stackData)-2]
		topics := make([]common.Hash, size)
		for i := 0; i < size; i++ {
			topic := stackData[len(stackData)-2-i]
````
IMO that would simplify it a bit, I don't think `stackPop` adds a whole lot

Reply:
good point, but 
```
topic := stackData[len(stackData)-2-i]
````
should be:

```
topic := stackData[len(stackData)-2-(i+1)]
```